src/gui/folderman.cpp

```diff
@@ -218,13 +218,27 @@ void FolderMan::setupFoldersHelper(QSettings &settings, AccountStatePtr account,
                 socketApi()->slotUnregisterPath(folderAlias);
                 auto settings = account->settings();
 
+                auto journalFileMoveSuccess = true;
+                // Due to db logic can't be sure which of these file exist.
+                if (oldJournal.exists()) {
+                    journalFileMoveSuccess = oldJournal.rename(folderDefinition.journalPath);
```

Comment:
Should be `&=` and not `=` here and in the other ifs. Otherwise you would miss failed moves as long as the last one succeeded.